### ISSUE TITLE
Kotlin/JS 2.0.0-RC1 Incompatibilities (environmental variables not working, error output not printed)

### ISSUE THREAD

**Issue by mgroth0** [kotest/kotest#3978]
Hello, aside from the lack of support for nested tests what is the current status for support of Kotlin/JS tests for Kotlin-2.0.0 RC1?

Here is my current experience in my real project.

First, these are some of the important JS configurations I have set:

-   **ecmaTarget**: es5 (not es2015)
-   **variant**: CommonJS (not ECMAScript)
-   **IR Compilation**
-   **Mocha Test Task**
-   **JS Target**: Browser

And I am doing this with versions:

-   Regular Kotest dependencies: `5.9.0.1440-SNAPSHOT`
-   Gradle plugin: `5.8.1` (I see no gradle plugin snapshot build)
-   Setting `kotestCompilerPluginVersion` to `5.9.0.1440-SNAPSHOT`

And my experience in my real project is the I see two issues, and unsure if they are related or not:

1. Error output is not printed. I just see

```txt
command '/Users/matthewgroth/.gradle/nodejs/node-v20.10.0-darwin-arm64/bin/node' exited with errors (exit code: 1)
```

Errors do not show in gradle output at all, even at the `--info` log level. I have to copy and paste the full test shell command to a terminal in order to see any errors.I saw a commit from 3 weeks ago that mentioned fixing log output: <https://github.com/kotest/kotest/commit/2d62856f6bb1cea575e6960dc62340d41a58c3af>But I am using the latest snapshot. I'm not sure if that commit was applied in the snapshot I am using?

2. when the kotest plugin is applied for some reason, environmental variables are no longer passed to tests.

(this property is broken: [https://github.com/JetBrains/kotlin/blob/d19d5c38feb8cc88064235b322e3dc44d3cfc32f/[...]/org/jetbrains/kotlin/gradle/targets/js/testing/KotlinJsTest.kt](https://github.com/JetBrains/kotlin/blob/d19d5c38feb8cc88064235b322e3dc44d3cfc32f/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/js/testing/KotlinJsTest.kt#L44))

Please let me know if I should split this into two different issues.

**Comment by OliverO2** [comment]
> 1. Error output is not printed. I just see

This is about a Node.js test, right? If so, the issue is #3329, which has been around for a while but will get fixed by #3954 once that is merged. I'll try to get this done by next weekend, so then there will be a snapshot you could use.

> 2. when the kotest plugin is applied for some reason, environmental variables are no longer passed to tests.

I have no idea. Are you expecting environment variables to be passed to JS in the browser? In any case, please open a separate issue and refer to relevant documentation.

**Comment by mgroth0** [comment]
It's been a while and issue 2 is likely no longer relevant to me so I probably won't be following up on it.

Since the PR you mentioned was merged, feel free to close this. I have since changed a lot in my project and don't use kotest for JS tests so I can't reproduce the issue any more, but since you merged that PR I'm ok with assuming its fixed.
